### PR TITLE
check for client before calling clientoptions

### PR DESCRIPTION
**Pull request by amiraabouhadid**
### Changes

- fixed if conditions to not read null 

### Related Issues

#3411 

### Tested Scenarios

- tested when logged in and after logging out
- tested that deployment timeout isn't affected

### CODE REVIEW

File: packages/playground/src/App.vue [hunk 1]
```diff
@@ -336,16 +336,19 @@ async function setTimeouts() {
...
+    if (clientOptions) {
+      const deploymentTimeoutMinutes = clientOptions.deploymentTimeoutMinutes;
+      if (!localStorageDeploymentTimeout && deploymentTimeoutMinutes) {
+        localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
+      } else {
+        client.clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
+        await client.connect();
+      }
     }
   }
```
Comment (zaelgohary) [inline packages/playground/src/App.vue:350]:
```suggestion
  if (client && client.clientOptions) {
    let deploymentTimeoutMinutes = client.clientOptions.deploymentTimeoutMinutes;
    if (!localStorageDeploymentTimeout && deploymentTimeoutMinutes) {
      localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
    } else {
      deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
      await client.connect();
    }
  }
```

Comment (zaelgohary) [reply packages/playground/src/App.vue:350]: I tested this by logging then navigating to any page then loggin out. Works fine.

File: packages/playground/src/App.vue [hunk 2]
```diff
@@ -336,16 +336,19 @@ async function setTimeouts() {
...
-      await client.connect();
+
+  if (client) {
+    const clientOptions = client.clientOptions;
+    if (clientOptions) {
+      const deploymentTimeoutMinutes = clientOptions.deploymentTimeoutMinutes;
+      if (!localStorageDeploymentTimeout && deploymentTimeoutMinutes) {
+        localStorage.setItem(LocalStorageSettingsKey.TIMEOUT_DEPLOYMENT_KEY, `${+deploymentTimeoutMinutes * 60}`);
+      } else {
+        client.clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
```
Comment (maayarosama) [inline packages/playground/src/App.vue]:
```suggestion
        clientOptions.deploymentTimeoutMinutes = +localStorageDeploymentTimeout! / 60;
```
you already have a clientOptions variable that you can use directly

**Review by maayarosama — DISMISSED** [review]

File: packages/playground/src/App.vue (hunk 2 above)
Comment (amiraabouhadid) [reply packages/playground/src/App.vue]: doesn't update the timeout on the client. Note how I was still able to deploy here even though timeout is 3 seconds. 
Screenshot from 2024-09-21 21-34-16

**Comment by amiraabouhadid** [comment]
@zaelgohary your suggestion fails to update the timeout on the client. I was able to deploy even though deploymentTimeout is set to 3 seconds in localstorage. 
Screenshot from 2024-09-21 21-34-16
Found issue in your suggestion and fixed. deploymentTimeout needs to be set on the client not a variable taken from a client.

**Review by zaelgohary — APPROVED** [review]

**Review by maayarosama — APPROVED** [review]

**Review by samaradel — APPROVED** [review]

**Pull request merged**